### ISSUE TITLE
Setting PS1 after posix shell initialisation breaks directory registering

### ISSUE THREAD

**Issue by kalocsaibotond** [ajeetdsouza/zoxide#939]
Having a shell startup command like
```sh
...
eval "$(zoxide init posix --hook prompt)"
...
PS1='[\u@\h \W]\$'
...
```
for a posix shell will break zoxide's directory registering capability (for zoxide 0.9.6 ). I tried this out for bash 5.2.32(1) and dash 0.5.12_2 . This is not the case for bash initialisation. After `eval "$(zoxide init bash)"` setting PS1 will not interfere with zoxide.

I surmise, hence the requirement to put zoxide initialisation as the last startup command.

Nevertheless, this is problematic if I want to set zoxide to be a system-wide available tool, since conventionally, I have to initialise zoxide somewhere in `/etc` .
If a user sets PS1 then that user inadvertently breaks his access to zoxide.

**Comment by ajeetdsouza** [comment]
Yes, the README mentions initializing zoxide last for exactly this reason. Is there an alternate solution you're proposing?

**Comment by kalocsaibotond** [comment]
At the moment, I do not have a better idea for the implementation. However, since zoxide explicitly use the contents of PS1, zoxide ought to raise a warning, when the contents of PS1 is not adequate for proper operation.  So for example zoxide ought to raise a warning, after using z or zi.
Having such a warning, would notify the user how zoxide use PS1, and hence the user can bypass the problem, via setting the PS1 in such a way to preserve zoxide's part (for example with sed-ing out the shell default part of PS1).

**Comment by ajeetdsouza** [comment]
Added: https://github.com/ajeetdsouza/zoxide/commit/1d08fe2b84772aadbfddf48fd590231d48511673
